### PR TITLE
add groupie to display list of device information

### PR DESCRIPTION
**Pull request by ShaunPlummer**
This PR refactors the app to use [groupie](https://github.com/lisawray/groupie), to manage the recycler view. Has a lot of built-in features for managing items as groups, handling updates and working with different kinds of items. 

At its core each item we would like to display should be a class extending from `Item`, allowing it to specify the layout used to represent it with `getLayout()` and `bind` to update a view element on request. I've included `SimpleListItem` as a simple base class for items to optionally extend from and updated `ManufacturerData` to make use of it.

In the future, the `value` param could be changed to `Any` to make this even more flexible. This contributes to issue #5 

```
open class SimpleListItem(private val label: Int, private val value: String? = "") : Item() {

    override fun getLayout() = R.layout.item_key_value

    override fun bind(viewHolder: GroupieViewHolder, position: Int) {
        viewHolder.containerView.context.apply {
            viewHolder.firstLabel.text = getString(label)
            viewHolder.firstValue.text = value
        }
    }
}
```

image

### CODE REVIEW

**Comment by ShaunPlummer** [comment]
The import order needs to be changed. This should probably be merged after pull request #14

**Comment by ShaunPlummer** [comment]
@Chesire  based on your comment on [this PR](https://github.com/b-reynolds/device-info-app/pull/12) I've added my own environment variable for the missing token.

I don't think I can re-run the workflow, at least if I can I can't find it :sweat:

**Comment by Chesire** [comment]
PR #14 is now merged in, should be able to do a rebase now

File: app/build.gradle [hunk 1]
```diff
@@ -1,55 +1,61 @@
-apply plugin: 'com.android.application'
```
Comment (Chesire) [inline app/build.gradle:1]: Any idea why it said the whole file changed? I guess maybe line endings or something?

File: build.gradle [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
-    }
-}
-
-task clean(type: Delete) {
-    delete rootProject.buildDir
-}
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+buildscript {
+    ext.kotlin_version = "1.4.10"
+    ext.groupie_version = "2.8.1"
```
Comment (Chesire) [inline build.gradle:4]: is this needed in here as opposed to the modules build.gradle?

**Comment by Chesire** [comment]
Is the line endings incorrectly configured on your git? Looks like all the touched files completely changed

**Comment by Chesire** [comment]
> @Chesire based on your comment on [this PR](https://github.com/b-reynolds/device-info-app/pull/12) I've added my own environment variable for the missing token.
> 
> I don't think I can re-run the workflow, at least if I can I can't find it 😓

It looks like we need to expose the Danger api token in some way, as it is still failing :/

File: app/build.gradle (hunk 1 above)
Comment (ShaunPlummer) [reply app/build.gradle]: Now fixed. I've converted the file back to windows line endings.

File: build.gradle (hunk 2 above)
Comment (ShaunPlummer) [reply build.gradle]: I've moved it to the module gradle.

**Comment by ShaunPlummer** [comment]
> Is the line endings incorrectly configured on your git? Looks like all the touched files completely changed

Now fixed. I've converted the file back to windows line endings.

> It looks like we need to expose the Danger api token in some way, as it is still failing :/
I'm a little confused about how it works. The Github actions workflow doesn't exist in my fork, only the main repo.

**Review by Chesire — APPROVED** [review]

**Pull request merged**